README.md

```diff
@@ -0,0 +1,82 @@
+## Design proposal: process manager API
+
+This document outlines design proposal for a prototype of a service API to manage Linux processes.
+
+The service is implemented as a server, allowing multiple concurrent client connections.
+
+### Library
+
+The library is the core of the server, which does the following:
+ - exposes gRPC endpoint with mTLS
+ - implements protos and API calls for
+   - starting a process
+   - stopping a process
+   - getting process status
+   - streaming process output
+ - performs API authorization
+
+The server maintains a map of processes, both active and stopped. The key is a UID generated by the server (not to be confused with PID assigned by the OS). The values is a structure represents the process.
+
+### API implementation
+
+`start`:
+ - Input: executable name and optional list of arguments.
+ - Output: process UID.
+ - Action: The server generates a UID, starts the process, and adds a new entry to the process map. The process output is forwarded to a file.
+
+`stop`:
+ - Input: process UID.
+ - Output: process status.
+ - Action: if the corresponding process is running, terminate the process and return status `stopped`. If the process is not in the map, return status `not found`.
+
+`get-status`:
+ - Input: process UID.
+ - Output: process status.
+ - Action: if the process is in the map, return process status. Otherwise return status `not found`.
+
+`stream-output`:
```

Comment:
Could you add details on how streaming will work? How will clients get real-time logs? And how will multiple clients be supported?